src/main/java/net/sf/jabref/model/metadata/MetaData.java

```diff
@@ -173,139 +92,117 @@ public AbstractBibtexKeyPattern getBibtexKeyPattern(GlobalBibtexKeyPattern globa
-            data.add(bibtexKeyPattern.getDefaultValue().get(0));
-            this.putData(KEYPATTERNDEFAULT, data);
+            defaultCiteKeyPattern = defaultValue.get(0);
         }
 
-        this.bibtexKeyPattern = bibtexKeyPattern;
+        postChange();
     }
 
-    public Optional<List<String>> getSaveActions() {
-        return Optional.ofNullable(getData(SAVE_ACTIONS));
+    public Optional<FieldFormatterCleanups> getSaveActions() {
+        return Optional.ofNullable(saveActions);
     }
 
     public Optional<BibDatabaseMode> getMode() {
-        List<String> data = getData(DATABASE_TYPE);
-        if ((data == null) || data.isEmpty()) {
-            return Optional.empty();
-        }
-        return Optional.of(BibDatabaseMode.parse(data.get(0)));
+        return Optional.ofNullable(mode);
     }
 
     public boolean isProtected() {
-        List<String> data = getData(PROTECTED_FLAG_META);
-        if ((data == null) || data.isEmpty()) {
-            return false;
-        } else {
-            return Boolean.parseBoolean(data.get(0));
-        }
+        return isProtected;
     }
 
     public Optional<String> getDefaultFileDirectory() {
-        List<String> fileDirectory = getData(FILE_DIRECTORY);
-        if ((fileDirectory == null) || fileDirectory.isEmpty()) {
-            return Optional.empty();
-        } else {
-            return Optional.of(fileDirectory.get(0).trim());
-        }
+        return Optional.ofNullable(defaultFileDirectory);
     }
 
     public Optional<String> getUserFileDirectory(String user) {
-        List<String> fileDirectory = getData(FILE_DIRECTORY + '-' + user);
-        if ((fileDirectory == null) || fileDirectory.isEmpty()) {
-            return Optional.empty();
-        } else {
-            return Optional.of(fileDirectory.get(0).trim());
-        }
+        return Optional.ofNullable(userFileDirectory.get(user));
     }
 
-    public Map<String, List<String>> getMetaData() {
-        return new HashMap<>(metaData);
-    }
-
-    public void setSaveActions(List<String> actionsSerialized) {
-        putData(SAVE_ACTIONS, actionsSerialized);
+    public void setSaveActions(FieldFormatterCleanups saveActions) {
+        this.saveActions = Objects.requireNonNull(saveActions);
+        postChange();
     }
 
     public void setSaveOrderConfig(SaveOrderConfig saveOrderConfig) {
-        List<String> serialized = saveOrderConfig.getAsStringList();
-        putData(SAVE_ORDER_CONFIG, serialized);
+        this.saveOrderConfig = saveOrderConfig;
+        postChange();
     }
 
     public void setMode(BibDatabaseMode mode) {
-        putData(DATABASE_TYPE, Collections.singletonList(mode.getAsString()));
+        this.mode = Objects.requireNonNull(mode);
+        postChange();
     }
 
     public void markAsProtected() {
-        putData(PROTECTED_FLAG_META, Collections.singletonList("true"));
+        isProtected = true;
+        postChange();
```

Comment:
So the change notification I asked for earlier have been moved into the meta data. Am I right?